Always show 3DS titles + Improve hardcoded error messages

simonj0 opened the pull request:
Summary of the changes:
- Always show 3DS titles + Disable 3DS downloads
  - A valid 3DS titlekeys url is automatically supplied if the WiiU url is valid, the 3DS input field and label in the ticket form is disabled/hidden
  - The custom response on `3ds.titlekeys.gq/json` is generated from games3ds.json, updates3ds.json and dlcs3ds.json (games3ds.json was regenerated to include names, which were empty before)
  - 3 Patches are implemented to prevent queueing of 3DS titles/updates/DLCs, 2 additional patches for skipping the download of system archives when launching a 3DS game
- Improve error messages
  - 2 Patches are implemented to replace the hardcoded error messages in the titlekey form and the ticket fetching dialog with more detailed error messages
  - WebExceptions in the `/zipProxy.php` handler are forwarded to the client instead of silently timing out
- Fix race condition in injector
  - In some cases the program started before all important patches were applied; the module initializer now waits for all patches to be applied before launching the main program

FailedShack commented:
LGTM 👍

simonj0 commented:
Added another commit for handling timeouts as well.

FailedShack commented on USBHelperLauncher/Net/ApplicationEndpoint.cs:
```diff
@@ -29,26 +30,50 @@ public void GetZipProxy(Session oS)
-            string content;
-            using (WebClient client = new WebClient())
+
+            try
```
Since this try catch only concerns `client.DownloadData(url);` it can be shrinked to contain that single line.

FailedShack commented on USBHelperLauncher/Net/ApplicationEndpoint.cs:
```diff
@@ -29,26 +30,50 @@ public void GetZipProxy(Session oS)
+                Proxy.LogRequest(oS, this, "Unable to create zip for " + url + ": " + webEx);
+
+                oS.utilCreateResponseAndBypassServer();
+                oS.utilSetResponseBody("");
```
Not needed, `utilCreateResponseAndBypassServer();` already sets an empty response.

FailedShack commented on USBHelperLauncher/Net/ApplicationEndpoint.cs:
```diff
@@ -29,26 +30,50 @@ public void GetZipProxy(Session oS)
+
+                oS.utilCreateResponseAndBypassServer();
+                oS.utilSetResponseBody("");
+                if (webEx.Response is HttpWebResponse)
```
I checked the docs and apparently `Response` is actually null when no response is available so a null check would suffice.
However, I noticed that the example given in the same page does this by checking if `e.Status == WebExceptionStatus.ProtocolError`. We should use that instead.

FailedShack commented:
Oh, and `SendInjectorSettings()` should be converted from `Task` to `void`.

simonj0 commented:
I've fixed the things you mentioned, and also modified the WebException messages to reflect the errors from Fiddler more accurately (instead of showing the localized description given by the status code)

simonj0 commented:
Fixed a potential issue in the last commit, should be fine now.
